library/std/src/sys/unix/thread_local_dtor.rs

```diff
@@ -11,28 +11,47 @@
 // Note, however, that we run on lots older linuxes, as well as cross
 // compiling from a newer linux to an older linux, so we also have a
 // fallback implementation to use as well.
+#[allow(unexpected_cfgs)]
 #[cfg(any(target_os = "linux", target_os = "fuchsia", target_os = "redox"))]
 pub unsafe fn register_dtor(t: *mut u8, dtor: unsafe extern "C" fn(*mut u8)) {
     use crate::mem;
     use crate::sys_common::thread_local_dtor::register_dtor_fallback;
 
+    /// This is necessary because the __cxa_thread_atexit_impl implementation
+    /// std links to by default may be a C or C++ implementation that was not
+    /// compiled using the Clang integer normalization option.
+    #[cfg(not(sanitizer_cfi_normalize_integers))]
+    #[cfi_encoding = "i"]
```

Comment:
why the explicit encoding? i would expect `c_int` to be an integer already as its `repr(transparent)`.

Reply:
This is the encoding of the C `int` type, which from a CFI perspective is distinct from `i32` or any other rust type. `int` and `long` can be the same size and in that case would be the same rust type, but CFI does distinguish between them unless `-Zcfi-normalize-integers` is used to only record the integer type size rather than it's name.